### PR TITLE
Implement user email verification on FE

### LINKED ISSUE #1528: Email verification link throws a 404 error
**Describe the bug**

Email verification link sent on email throws a 404 error.

**To Reproduce**
1. Create a user
2. Click on the verification link that is sent on the provided email

**Expected behavior**

User should be verified successfully.

**Screenshots**

screen shot 2018-07-30 at 3 41 18 pm

Working on fixing this issue.

### PR DESCRIPTION
**Pull request by schedutron**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
This PR adds user email verification via front-end. Here is the screenshot after successful email verification:

screen shot 2018-07-31 at 7 11 12 am

Fixes #1528

### CODE REVIEW

**Comment by pradeepgangwar** [comment]
@schedutron please fix the linting errors

File: app/routes/verify.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    };
+
+    return this.get('loader')
+      .post('auth/verify-email', payload)
+      .then(() => {
+        this.controllerFor('verify').message = 'Your email has been verified successfully! Please login to continue.';
+        this.controllerFor('verify').success = true;
+      })
+      .catch(() => {
+        this.controllerFor('verify').message = 'An unexpected error occured.';
```
Comment (srv-twry) [inline app/routes/verify.js]: Don't say that an unexpected error occured. Show them the real error message response sent from the server.

File: app/routes/verify.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let payload = {
+      data: {
+        token: transition.queryParams.token
+      }
+    };
+
+    return this.get('loader')
+      .post('auth/verify-email', payload)
+      .then(() => {
+        this.controllerFor('verify').message = 'Your email has been verified successfully! Please login to continue.';
```
Comment (CosmicCoder96) [inline app/routes/verify.js]: This is going against the ember design principles. First of all, this entire logic should be an action inside the controller. 
Secondly, the approach is very hacky, you are using a route to manipulate properties of it's controller which are available readily inside the controller.
Even after shifting the property to the controller, keep all the message text inside the template itself and use `if-else` handlebar helpers to decide which message to display.

Comment (schedutron) [reply app/routes/verify.js]: After moving the code to the `verify` controller, how do I trigger it automatically after the verification link is clicked?

**Pull request merged**